woocommerce/woocommerce-ios
# REST API: Check passwordless account for the input WPCom email

Part of #8918 
⚠️ Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/746.


## Description

This PR integrates the WPCom account check from the WordPressAuthenticator library when an email is entered in the email login screen of the new Jetpack setup flow. 

In order to use the public method `requestAuthenticationLink` of the library, the authenticator needs to be initialized with configs - so I moved the initialization from `AuthenticationManager` to an extension of `WordPressAuthenticator` so that it can be reused from the Jetpack setup flow too.

Also, more tests for `JetpackSetupCoordinator` were added.

## Testing instructions

Prerequisite: Make sure that you have access to a site with WooCommerce and no Jetpack.
- Log out of the app or skip onboarding if needed.
- Tap "Enter your site address" on the prologue screen and proceed with the address of your test site.
- Log in with the credentials of an admin.
- When the login completes, tap the Jetback benefit banner at the bottom of the Dashboard screen.
- Tap Install Jetpack on the benefit modal.
- Enter any email address on the presented WPCom email login screen.
- Notice that the CTA shows a loading indicator when the account check is being requested, and stops when done. 
- Notice in Xcode console either `✅ account check done - passwordless: true/false` or `⛔️ Error checking for passwordless account:<error>`. The response will be handled in the next PRs.

## Screenshots


https://user-images.githubusercontent.com/5533851/221842046-604b900c-198e-4f32-aa04-83fc6da11961.mp4



---
